scripts/missions/soa/4_6_0_Balamors_Ruse.lua

```diff
@@ -16,16 +16,17 @@ mission.reward =
 local keyItemOnMobDeath =
 {
     onMobDeath = function(mob, player, optParams)
-        -- TODO: Mobs that grant this KI and the rate in which it drops needs to be measured further.  This
-        -- is currently set to a lower value than observed in capture.
-
-        if math.random(1, 100) <= 20 then
-            local playerZoneID = player:getZoneID()
+        -- TODO: Mobs that grant this KI and the rate in which it drops needs to be measured further.
+        if
+            (optParams.isKiller or optParams.noKiller) and -- Only run once, not once per party member
+            math.random(1, 100) <= 20                      -- This is currently set to a lower value than observed in capture.
+        then
+            local zoneID = mob:getZoneID() -- Fetch the zoneId of the mob to ensure correct zone in case of no killer.
 
             for _, partyMember in ipairs(player:getParty()) do
```

Comment:
Don't want to use `player` and `noKiller` together since player will be nil if noKiller is true. If this is part of a battlefield you can pull players from that.